contracts/CDPLeverage.sol

```diff
@@ -0,0 +1,456 @@
+          uint daiAmountFinal,
+          uint256 totalDebt,
+          uint256 priceFloor,
+          uint256 index)
+    {
+       Investor memory investor = investors[_addr];
+       return (
+         investor.layers,
+         investor.principal,
+         investor.collatRatio,
+         investor.cdpID,
+         investor.purchPrice,
+         investor.daiAmountFinal,
+         investor.totalDebt,
+         investor.priceFloor,
+         investor.index);
+    }
+
+
+    /**
+    * @dev Returns amount of PETH for a given amount of WETH at exchange rate
+    * @param _wethAmount - literal
+    */
+    function weth2peth(uint256 _wethAmount)
+        view
+        public
+        returns (uint pethAmount) {
+        pethAmount = rdiv(_wethAmount, wmul(tub.gap(), tub.per())) - 1;      // WAD
+    }
+
+
+    /**
+    * @dev Executes a market buy order and returns the amount of asset recieved
+    * @param _to - token getting
+    * @param _from - token giving
+    * @param _fromAmount - amount of token giving
+    */
+    function marketBuy(DSToken _to, DSToken _from, uint256 _fromAmount)
+        internal
+        returns (uint256 toAmount)
+    {
+        toAmount = dex.getBuyAmount(_to, _from, _fromAmount);                // calculate how much of _to to get at market and with _fromAmount
+        dex.buyAllAmount(_to, toAmount, _from, _fromAmount);                 // OasisDEX market buy
+    }
+
+
+    /**
+    * @dev Free's locked CDP collateral (PETH) and converts it to WETH
+    * @param _cdpID - verbatim
+    * @param _peth - amount of peth
+    */
+    function releaseWeth(bytes32 _cdpID, uint256 _peth) internal {
+        tub.free(_cdpID, _peth);                                             // empty all unlocked peth to this account
+        tub.exit(_peth);                                                     // convert PETH to WETH
+    }
+
+
+    /**
+    * @dev Decreases CDP debt by wiping DAI and returns the remaining debt
+    * @param _cdpID - verbatim
+    * @param _daiAmount - verbatim
+    * @param _remainingDebt - in DAI
+    */
+    function wipeDebt(bytes32 _cdpID, uint256 _daiAmount, uint256 _remainingDebt)
+        internal
+        returns (uint256 remainingDebt)
+    {
+        tub.wipe(_cdpID, _daiAmount);                                        // wipe off some of the the debt by paying back some of the Dai amount
+        remainingDebt = sub(_remainingDebt,_daiAmount);                                        // deduct from previous debt
+    }
+
+
+    /**
+    * @dev Calculates the governance and stability fee (in DAI) in terms of weth and mkr at the given market rate
+    * @param _cdpID - verbatim
+    * @param _remainingDebt - in DAI
+    */
+    function wipeFee(bytes32 _cdpID, uint256 _remainingDebt)
+        internal
+        returns (uint256 mkrFee, uint256 wethFee)
+    {
+        bytes32 val;
+        (val, ) = pep.peek();                                                // maker price feed
```

Comment:
If we're not using the returned boolean, we can just call `read` on pep the same way we do on pip (they're both DSValue contracts).

Reply:
Yes, indeed; I'll add it to the improvement list. This logic was used in tub.sol, which carries near zero uncertainty.